fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -3096,6 +3104,48 @@ public void createTableLike(CreateTableLikeStmt stmt) throws DdlException {
         }
     }
 
+    public void createTableAsSelect(CreateTableAsSelectStmt stmt) throws DdlException {
+        try {
+            QueryStmt tmpStmt = stmt.getQueryStmt().clone();
```

Comment:
Why not put this logic in `analyze()` of `CreateTableAsSelectStmt`?

Reply:
Ok, I'll modify it